pkg/cmd/server/server.go

```diff
@@ -60,15 +63,36 @@ import (
 )
 
 func NewCommand() *cobra.Command {
-	var kubeconfig string
+	var (
+		kubeconfig   string
+		logLevelFlag = newLogLevelFlag()
+	)
 
 	var command = &cobra.Command{
 		Use:   "server",
 		Short: "Run the ark server",
 		Long:  "Run the ark server",
 		Run: func(c *cobra.Command, args []string) {
-			logger := logrus.New()
-			logger.Hooks.Add(&logging.ErrorLocationHook{})
+			var (
+				logLevel = logrus.InfoLevel
+				parsed   logrus.Level
+				err      error
+			)
+
+			if parsed, err = logrus.ParseLevel(logLevelFlag.String()); err == nil {
+				logLevel = parsed
+			}
+
+			logger := newLogger(logLevel, &logging.ErrorLocationHook{}, &logging.LogLocationHook{})
+
+			// This is the log-level parse error from above. This should theoretically
+			// never happen assuming the enum flag is constructed correctly because the
+			// enum flag will not allow an invalid value to be set.
+			if err != nil {
+				logAtMinLevel(logger, "log-level flag has invalid value %s; setting log-level to %s", strings.ToUpper(logLevelFlag.String()), strings.ToUpper(logLevel.String()))
```

Comment:
This is clever but can we just assume that this is an error and log it as such but let execution continue?

Reply:
+1, no need for logAtMinLevel. Also, as a follow-up to my comment above, I would just put this in an `else` block when parsing, and just do `logrus.Errorf`